routes/articles.js

```diff
@@ -10,4 +10,15 @@ router.get('/', (req, res) => {
     })
 })
 
+router.post('/article', (req,res) => {
+    const db =req.app.get('db')
+    const title = req.body.title
+    const body = req.body.body
+    const author = req.body.author
+    db.query('insert into articles(??,??,??)',[title,body,author], (err) => {
```

Comment:
This SQL statement will not work. `articles` has 4 columns and you're trying to insert 3, so it'll give an error. What you need to do is specify which columns you want to insert it, and the if the others are not set to `NOT NULL`, it will work.
````
INSERT INTO articles (title, body, author) VALUES ("Sample Title", "Some body 👀 ", "Author's Name")
````
`id` will be auto incremented as we set in the schema.

The way you're making the prepared query _(the `??`)_ is not right. Check [the docs](https://github.com/mysqljs/mysql#preparing-queries). To generate the SQL statement, you need to call a function (`mysql.format()`) with the first string as the SQL query with `??`s, and the second parameter is a list of values / variables to replace the `??`s.

Reply:
Fixed